src/extensions/metadata.rs

```diff
@@ -0,0 +1,172 @@
+//! Adds support for the IMAP METADATA command specificed in [RFC5464](https://tools.ietf.org/html/rfc5464).
+
+use std::fmt::Debug;
+
+use async_std::io::{Read, Write};
+use async_std::prelude::*;
+use async_std::stream::Stream;
+use async_std::sync;
+use imap_proto::types::{MailboxDatum, Metadata};
+use imap_proto::{RequestId, Response};
+
+use crate::client::{validate_str, Session};
+use crate::codec::ResponseData;
+use crate::error::Result;
+use crate::parse::handle_unilateral;
+use crate::types::UnsolicitedResponse;
+
+trait CmdListItemFormat {
```

Comment:
Given that this trait has only one implementation, it doesn't seem to be helpful, especially if it is a private trait